### ISSUE TITLE
pygrambank warnings on UraTyp-sheets

### ISSUE THREAD

**Issue by HedvigS** [bedlan/uratyp-1#3]
```
Estonian
RK-MN_esto1258	ERROR	?		GB260 must have a comment if it is coded 1

Livvi_Karelian
RK-MN_livv1243	ERROR	?		GB260 must have a comment if it is coded 1

Tver_Karelian
RK-MN_tver1240	ERROR	?		GB260 must have a comment if it is coded 1

Courland_Livonian
RK-MN_west1760	ERROR	?		GB260 must have a comment if it is coded 1

East_Mansi
RK_east2879	ERROR	78	GB124	value without source
RK_east2879	ERROR	168	GB325	value without source
RK_east2879	ERROR	194	GB520	value without source

Komi_Zyrian
RK_komi1268	ERROR	?		GB260 must have a comment if it is coded 1
RK_komi1269	ERROR	?		GB260 must have a comment if it is coded 1

Soikkola_Ingrian
RK_soyk1238	ERROR	?		GB260 must have a comment if it is coded 1

Voro_South_Estonian
RK_voro1241	ERROR	?		GB260 must have a comment if it is coded 1

West_Votic
RK_west2391	ERROR	?		GB260 must have a comment if it is coded 1
```

**Comment by xrotwang** [comment]
UraTyp aims to supply each value of 1 with an example, which presumably can stand in for a comment.

**Comment by HedvigS** [comment]
> UraTyp aims to supply each value of 1 with an example, which presumably can stand in for a comment.

Sure. If @miinaNO and the team want to use pygrambank in their workflow (#8 ) then we can adjust it to accept a non-missing example value for missing comments.

**Comment by HedvigS** [comment]
I don't know why this is tagged as completed. There are still warnings about missing sources.

**Comment by HedvigS** [comment]
I believe that in this thread these issues remain

```

East_Mansi
RK_east2879	ERROR	78	GB124	value without source
RK_east2879	ERROR	168	GB325	value without source
RK_east2879	ERROR	194	GB520	value without source
```

**Comment by HedvigS** [comment]
Looking at the East Mansi sheet in full, none of the datapoints have a source. I actually don't know why pygrambank only flagged these three. Any ideas?

**Comment by HedvigS** [comment]
Johannes Englisch is not in this repos, otherwise I'd tag him here for his help.

**Comment by xrotwang** [comment]
As far as I can tell, the East Mansi sheet was never edited after I seeded it from Grambank three years ago. So I guess, either @MiinaNo hasn't pushed an updated version of the sheet, or we might just recreate it from Grambank, adding the Source column this time.
```
$ git log raw/GB/language-tables/East_Mansi.csv 
commit b3c7bcefde9ae01ab2a25ca052f5fdf9c307c023
Date:   Thu Dec 16 13:20:19 2021 +0100

    created per-language tables for GB from CLDF data
```

**Comment by xrotwang** [comment]
There's a couple more sheets without a `Source` column - possibly reflecting the state of the sheets in Grambank at that point in time.

**Comment by HedvigS** [comment]
Okay. 

What surprises me is that when we did the UraTyp -> GB merge last summer, there weren't more warnings about sources missing from pygrambank for East Mansi. I guess it's because I set it to only import values that aren't in Grambank, and that was only these three?

We discussed importing comments and sources from GB into UraTyp here #9 , but that seems to have been decided against, or at least dropped for now?

**Comment by xrotwang** [comment]
> We discussed importing comments and sources from GB into UraTyp here [#9](https://github.com/bedlan/uratyp-1/issues/9) , but that seems to have been decided against, or at least dropped for now?

Up to @MiinaNo to decide. She's the one to know whether "old" comments or sources would make sense considering her editing of most sheets.

**Comment by xrotwang** [comment]
Btw.: I remember that the files in `raw/GB` were seeded based on the value matrix used in the publication - hence no Source column.

**Comment by HedvigS** [comment]
> Btw.: I remember that the files in `raw/GB` were seeded based on the value matrix used in the publication - hence no Source column.

Yes, I also think that. That's why I think we started #9.

**Comment by MiinaNo** [comment]
> There's a couple more sheets without a `Source` column - possibly reflecting the state of the sheets in Grambank at that point in time.

Yes, that's true. There are a few sheets missing the `Source` column. These are the sheets we have not touched so far, thus, they reflect the earliest stage. It is fine to add the `Source` column if they are causing errors. The aim is to have a proper source at one point. In fact, I should be able to update some of these very old tables but let's see.

**Comment by MiinaNo** [comment]
> I believe that in this thread these issues remain
> 
> ```
> 
> East_Mansi
> RK_east2879	ERROR	78	GB124	value without source
> RK_east2879	ERROR	168	GB325	value without source
> RK_east2879	ERROR	194	GB520	value without source
> ```

@HedvigS it was fine to change these three ? to 0. for the time being. I approved the pull request.

**Comment by HedvigS** [comment]
> > There's a couple more sheets without a `Source` column - possibly reflecting the state of the sheets in Grambank at that point in time.
> 
> Yes, that's true. There are a few sheets missing the `Source` column. These are the sheets we have not touched so far, thus, they reflect the earliest stage. It is fine to add the `Source` column if they are causing errors. The aim is to have a proper source at one point. In fact, I should be able to update some of these very old tables but let's see.

As far as I understand based on what @xrotwang has said, this dataset was originally created just based on the Value fields. It was not chosen to include the source field from the start. I don't know what the dataset looked like before it was made in this form. Is there anyway to retrieve the original UraTyp source information form before CLDF?

**Comment by HedvigS** [comment]
> > I believe that in this thread these issues remain
> > ```
> > 
> > East_Mansi
> > RK_east2879	ERROR	78	GB124	value without source
> > RK_east2879	ERROR	168	GB325	value without source
> > RK_east2879	ERROR	194	GB520	value without source
> > ```
> 
> [@HedvigS](https://github.com/HedvigS) it was fine to change these three ? to 0. for the time being. I approved the pull request.

I'm a bit confused. There is not active PR in this repos and the sheet in the `uratyp-1/raw/GB/language-tables/East_Mansi.csv` still has ? for these values.

I think, setting aside East Mansi specifically for now, it would be good to know if the original source information from the UraTyp project can be retrieved.

**Comment by HedvigS** [comment]
(wait, @MiinaNo must mean #15 . That was indeed merged. What was happening was a fork problem, I was looking at Hedvigs/Uratyp-1 instead of bedland/Uratyp-1. I'll delete that fork immediately, that's just gonna be more confusion.)

**Comment by HedvigS** [comment]
When running pygrambank checks over UraTyp import into Grambank (https://github.com/glottobank/Grambank/pull/2949), we get these warnings. I'm going to ignore these for now, but I'll add them to this issue thread for records.

```
> MN-IE_akka1237      ERROR   ?               GB260 must have a comment if it is coded 1
> MN-IE_ters1235      ERROR   ?               GB260 must have a comment if it is coded 1
> RK-MN_esto1258      ERROR   ?               GB260 must have a comment if it is coded 1
> RK-MN_tver1240      ERROR   ?               GB260 must have a comment if it is coded 1
> RK-MN_west1760      ERROR   ?               GB260 must have a comment if it is coded 1
> RK_east2879         ERROR   168     GB325   value without source
> RK_komi1268         ERROR   ?               GB260 must have a comment if it is coded 1
> RK_komi1269         ERROR   ?               GB260 must have a comment if it is coded 1
> RK_lule1254         ERROR   ?               GB260 must have a comment if it is coded 1
> RK_soyk1238         ERROR   ?               GB260 must have a comment if it is coded 1
> RK_voro1241         ERROR   ?               GB260 must have a comment if it is coded 1
> RK_west2391         ERROR   ?               GB260 must have a comment if it is coded 1
```
